src/mpid/ch4/generic/am/mpidig_am_part.h

```diff
@@ -49,7 +49,8 @@ MPL_STATIC_INLINE_PREFIX int MPIDIG_part_start(MPIR_Request * request)
 /* Checks whether all partitions are ready and MPIDIG_PART_REQ_CTS has been received,
  * If so, then issues data. Otherwise a no-op.
  */
-MPL_STATIC_INLINE_PREFIX int MPIDIG_post_pready(MPIR_Request * part_sreq)
+MPL_STATIC_INLINE_PREFIX int MPIDIG_post_pready(MPIR_Request * part_sreq,
+                                                enum MPIDIG_part_issue_mode mode)
```

Comment:
I think it was not a good design to have `post_pready` to serve both the `MPI_Pready` path and "CTS" reply path. Serving two purposes makes the name non-intuitive. Maybe we can refactor it into something like --
```
[MPI_Pready path]
update_pready_partitions
if (can_issue_data()) {
    issue_data(is_reply=false);
}

[CTS reply path]
update_cts_status()
if (can_issue_data()) {
    issue_data(is_reply=true);
}
```

Reply:
Sure, I can make that change.

Reply:
I did this as an additional commit, but can rebase or squash if we want to clean up the history some.